src/vs/workbench/parts/tasks/common/taskSystem.ts

```diff
@@ -85,6 +83,7 @@ export const enum TaskExecuteKind {
 export interface ITaskExecuteResult {
 	kind: TaskExecuteKind;
 	promise: TPromise<ITaskSummary>;
+	task: Task;
```

Comment:
Can you quickly explain why we need this?

Reply:
In `handleExecuteResult` in task.contribution.ts we need the Task that was executed. For tasks that were run using rerun, task.contribution.ts has no knowledge of the task. Putting the task in the execute result solves this.